### PR TITLE
Support for continuous joint type and Script to generate urdf_description ROS package

### PR DESCRIPTION
**Pull request by KamalanathanN**
URDF of wheeled platforms in ROS requires wheel joints to be continuous joint type.

### CODE REVIEW

File: onshape_to_robot/load_robot.py [hunk 1]
```diff
@@ -179,7 +180,7 @@ def assignParts(root, parent):
                 print(Fore.RED + 'ERROR: "' + name +
                       '" is declared as a DOF but the mate type is '+data['mateType']+'')
                 print(
-                    '       Only REVOLUTE, CYLINDRICAL, SLIDER and FASTENED are supported' + Style.RESET_ALL)
+                    '       Only REVOLUTE, CYLINDRICAL, SLIDER, CONTINUOUS and FASTENED are supported' + Style.RESET_ALL)
```
Comment (Gregwar) [inline onshape_to_robot/load_robot.py:183]: This warning refers to OnShape mates that are supported, and there is no such mate as CONTINUOUS

**Comment by Gregwar** [comment]
Hello, indeed switching limitless joints to continuous is a good feature (and doesn't break any BC I think)

File: onshape_to_robot/load_robot.py (hunk 1 above)
Comment (KamalanathanN) [reply onshape_to_robot/load_robot.py]: 😬 Ops.. Warning statement is corrected.

**Comment by KamalanathanN** [comment]
hi @Gregwar , yes it doesnt break any BC. I have tested it on my little wheeled robot model.
I've also created a script to automate urdf_description ROS package generation using onshpe-to-urdf. Kindly review and share your thoughts.

**Comment by Gregwar** [comment]
Hello,
Thanks for the details

If generating those extra files is a very common use case, we could consider having a flag in `onshape-to-robot` to add them automatically, but having an extra external script doesn't feel like a solution (maybe you can have an external "skeleton" repository)

(As well, a file is downloaded from your GitHub from that script)

File: big_bang.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+cp urdf/{*.stl,*.part} meshes/
+rm -r urdf/{*.stl,*.part}
+
+cd launch/
+
+wget https://raw.githubusercontent.com/KamalanathanN/onshape-to-robot/master/urdf.rviz
+
+cat >./display.launch <<EOF
+<launch>
```
Comment (rsinnet) [inline big_bang.sh:63]: I think Jinja2 would probably be a good solution for this rather than shell.

File: urdf.rviz [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        motor_2:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        n20-43mm-d-shaft-wheel-1.snapshot.9:
```
Comment (rsinnet) [inline urdf.rviz:84]: I don't think model-specific code would belong in this repo.

File: big_bang.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+set -e 
+
+pkg_name=$1
+echo "
+
+ ▄▄▄▄    ██▓  ▄████     ▄▄▄▄    ▄▄▄       ███▄    █   ▄████
```
Comment (rsinnet) [inline big_bang.sh:8]: LOL. Have you seen this tool? https://patorjk.com/software/taag/#p=display&f=Graffiti&t=Type%20Something%20

File: big_bang.sh [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+catkin_create_pkg ${pkg_name}_description rospy
+cd ${pkg_name}_description/
+mkdir launch meshes rviz urdf
+
+cd urdf/
+echo
+echo "Enter config details : "
+echo
+
+read -p 'documentId : ' doc_id
```
Comment (rsinnet) [inline big_bang.sh:32]: This interface doesn't seem to hide any complexity so, if I were the maintainer, I would have some concerns about code bloat.

**Review by rsinnet — COMMENTED** [review]
Just looking through this repo. Seeing what it will take to adapt to my use case.

**Pull request closed without merging**